src/DeviceInterface/InterfacePrototype.js

```diff
@@ -118,41 +128,75 @@ class InterfacePrototype {
         return this.#data.creditCards
     }
 
+    async startInit () {
+        this.addDeviceListeners()
+        await this.setupAutofill()
+        const event = new CustomEvent('InitComplete', {})
+        window.dispatchEvent(event)
+    }
+
     init () {
-        const start = () => {
-            this.addDeviceListeners()
-            this.setupAutofill()
-        }
         if (document.readyState === 'complete') {
-            start()
+            this.startInit()
         } else {
-            window.addEventListener('load', start)
+            window.addEventListener('load', () => {
+                this.startInit()
+            })
         }
     }
 
-    attachTooltip (form, input) {
+    attachTooltip (form, input, e) {
         form.activeInput = input
+        this.setActiveForm(form)
+
+        const inputType = getInputMainType(input)
+        const subtype = getInputSubtype(input)
+        window.addEventListener('pointerdown', () => this.removeTooltip(), {capture: true, once: true})
+        window.addEventListener('input', () => this.removeTooltip(), {once: true})
+        if (!isTopFrame && isApp) {
+            this.showTopTooltip(form, input, inputType, subtype, e)
+            return
+        }
 
         if (isMobileApp) {
             this.getAlias().then((alias) => {
                 if (alias) form.autofillEmail(alias)
                 else form.activeInput.focus()
             })
         } else {
+            const config = getInputConfigFromType(inputType)
+
+            // TODO this also needs to be moved to data before we can remove the fake form
             if (form.tooltip) return
 
-            form.tooltip = !isApp
-                ? new EmailAutofill(input, form, this)
-                : new DataAutofill(input, form, this)
+
+            // In extensions, the tooltip is centered on the Dax icon
+            const position = isApp ? input.getBoundingClientRect()
+                : getDaxBoundingBox(input)
+            // TODO check if we should pass in a position function instead to measure the current input position.
+
+            form.tooltip = isApp
+                ? new DataAutofill(config, subtype, position, this, form)
+                : new EmailAutofill(config, subtype, position, this, form)
             form.intObs.observe(input)
```

Comment:
This function is the last remaining piece to removing the DOM driven nature of the top autofill. We should be able to pass in pure data that topAutofill.js already has and render these tooltips.